### PR TITLE
[174] - Create a page to view other user profiles reviews

### PR DESCRIPTION
**Pull request by khasanrashidov**
### Where

[NL-174 Create a page to view other user profiles reviews](https://jjbcapital.atlassian.net/browse/NL-174)

### What

The goal of the ticket is to create a page where the current user could see the rating/reviews of other users.

### How

Implemented everything specified in the ticked. Now, user besides being able to look in the profile of other user, can look at the reviews/rates of that user. Every case was taken into account (page navigation (current user or other user), deleted user, user with no reviews).

### Affected Area

Auctionify.API.Controllers
Auctionify.API.ClientApp
Auctionify.Application.Features
Auctionify.UnitTests

### PR Checklist

_Build_

- [x] Build Succeeded

_Unit Tests_

- [x] All Tests Passed, 0 Ignored, 0 Failed

 Some screenshots:
 
image

image

### CODE REVIEW

**Review by otash8110 — APPROVED** [review]

File: src/Auctionify.Application/Features/Rates/Commands/AddRateToBuyer/AddRateToBuyerCommandValidator.cs [hunk 1]
```diff
@@ -43,58 +40,58 @@ ILotStatusRepository lotStatusRepository
...
+							cancellationToken: cancellationToken
+						);
 
 						var ratings = await _rateRepository.GetListAsync(
 								predicate: l => l.LotId == request.LotId,
 								cancellationToken: cancellationToken
 						);
 
-						if (ratings.Items.Count < 1 && !ratings.Items.Any(item => item.SenderId == lot.SellerId))
+						if (ratings.Items.Count <= 2 && !ratings.Items.Any(item => item.SenderId == lot.SellerId))
```
Comment (TsvirkoNikita) [inline src/Auctionify.Application/Features/Rates/Commands/AddRateToBuyer/AddRateToBuyerCommandValidator.cs:84]: Looks like rates count should less than 2 in this condition, isn't? If lot has 2 rates, that means that it already rated by seller and buyer. By the way, constants like that should be wrapped in variables with a telling name

File: src/Auctionify.Application/Features/Rates/Commands/AddRateToSeller/AddRateToSellerCommandValidator.cs [hunk 2]
```diff
@@ -83,7 +83,10 @@ ILotStatusRepository lotStatusRepository
 								cancellationToken: cancellationToken
 							);
 
-							if (ratings.Items.Count < 1 && !ratings.Items.Any(item => item.SenderId == lot.BuyerId))
+							if (
+								ratings.Items.Count <= 2
```
Comment (TsvirkoNikita) [inline src/Auctionify.Application/Features/Rates/Commands/AddRateToSeller/AddRateToSellerCommandValidator.cs:87]: Same as above

**Review by TsvirkoNikita — CHANGES_REQUESTED** [review]

File: src/Auctionify.Application/Features/Rates/Commands/AddRateToBuyer/AddRateToBuyerCommandValidator.cs (hunk 1 above)
Comment (khasanrashidov) [reply src/Auctionify.Application/Features/Rates/Commands/AddRateToBuyer/AddRateToBuyerCommandValidator.cs:84]: Yes, agree, I will declare them in constants, thank you for remark)

**Comment by khasanrashidov** [comment]
Fixed specified problems in comments.

**Comment by TsvirkoNikita** [comment]
> Fixed specified problems in comments.

Great, but what about the comparison condition? Shouldn't it be strictly less?
> Looks like rates count should less than 2 in this condition, isn't? If lot has 2 rates, that means that it already rated by seller and buyer.

**Review by SaidganiUsm — APPROVED** [review]

**Comment by khasanrashidov** [comment]
> > Fixed specified problems in comments.
> 
> Great, but what about the comparison condition? Shouldn't it be strictly less?
> 
> > Looks like rates count should less than 2 in this condition, isn't? If lot has 2 rates, that means that it already rated by seller and buyer.

Yes, I agree that first part of condition should be enough, ;et me think about this condition, this validation was written by @SaidganiUsm I just found one issue and made MaxRatingsPerLot equal 2, I will reply you in several minutes

**Comment by TsvirkoNikita** [comment]
> > > Fixed specified problems in comments.
> > 
> > 
> > Great, but what about the comparison condition? Shouldn't it be strictly less?
> > > Looks like rates count should less than 2 in this condition, isn't? If lot has 2 rates, that means that it already rated by seller and buyer.
> 
> Yes, I agree that first part of condition should be enough, let me think about this condition, this validation was written by @SaidganiUsm I just found one issue and made MaxRatingsPerLot equal 2, I will reply you in several minutes

No problem, but I didn't mean that only the first part of the condition is enough. I only meant to use < instead of <= in comparison with MaxRatingsPerLot

**Comment by khasanrashidov** [comment]
@TsvirkoNikita Alright, I got it. I studied the case and discussed with @SaidganiUsm . He told me that, actually, the check 

```
!ratings.Items.Any(item => item.SenderId == lot.BuyerId)
```
 or (for another handler validator case)

```
!ratings.Items.Any(item => item.SenderId == lot.SellerId)
```

is enough.

Yes, I agree with this, because it check that each individual user (buyer or seller in our case) have a rate for lot (as a receiver or sender).

In that case we don't need **MaxRatingsPerLot** variable. And the second check would be enough. What do you think @TsvirkoNikita

**Comment by TsvirkoNikita** [comment]
> @TsvirkoNikita Alright, I got it. I studied the case and discussed with @SaidganiUsm . He told me that, actually, the check
> 
> ```
> !ratings.Items.Any(item => item.SenderId == lot.BuyerId)
> ```
> 
> or (for another handler validator case)
> 
> ```
> !ratings.Items.Any(item => item.SenderId == lot.SellerId)
> ```
> 
> is enough.
> 
> Yes, I agree with this, because it check that each individual user (buyer or seller in our case) have a rate for lot (as a receiver or sender).
> 
> In that case we don't need **MaxRatingsPerLot** variable. And the second check would be enough. What do you think @TsvirkoNikita ? did we miss something?

Yep, got you, agree

**Comment by khasanrashidov** [comment]
Just to be sure I rechecked that in frontend and this condition really works:

image

**Comment by khasanrashidov** [comment]
Updating locally the branch, merging with dev

**Comment by khasanrashidov** [comment]
That's it this branch is good to be merged to dev.

**Review by TsvirkoNikita — APPROVED** [review]

**Review by SaidganiUsm — APPROVED** [review]

**Pull request merged**